src/main/scala/zio/ftp/Ftp.scala

```diff
@@ -1,121 +1,85 @@
-               )
-           )
-      data <- ZStream.fromInputStream(is, chunkSize).chunks
-    } yield data)
-
-  def rm(path: String): ZIO[Blocking, IOException, Unit] =
-    execute(_.deleteFile(path))
-      .filterOrFail(identity)(InvalidPathError(s"Path is invalid. Cannot delete file : $path"))
-      .unit
-
-  def rmdir(path: String): ZIO[Blocking, IOException, Unit] =
-    execute(_.removeDirectory(path))
-      .filterOrFail(b => b)(InvalidPathError(s"Path is invalid. Cannot delete directory : $path"))
-      .unit
-
-  def mkdir(path: String): ZIO[Blocking, IOException, Unit] =
-    execute(_.makeDirectory(path))
-      .filterOrFail(identity)(InvalidPathError(s"Path is invalid. Cannot create directory : $path"))
-      .unit
-
-  def ls(path: String): ZStream[Blocking, IOException, FtpResource] =
-    ZStream
-      .fromEffect(execute(_.listFiles(path).toList))
-      .flatMap(Stream.fromIterable)
-      .map(FtpResource(_, Some(path)))
-
-  def lsDescendant(path: String): ZStream[Blocking, IOException, FtpResource] =
-    ZStream
-      .fromEffect(execute(_.listFiles(path).toList))
-      .flatMap(Stream.fromIterable)
-      .flatMap { f =>
-        if (f.isDirectory) {
-          val dirPath = Option(path).filter(_.endsWith("/")).fold(s"$path/${f.getName}")(p => s"$p${f.getName}")
-          lsDescendant(dirPath)
-        } else
-          Stream(FtpResource(f, Some(path)))
-      }
-
-  def upload[R <: Blocking](path: String, source: ZStreamChunk[R, Throwable, Byte]): ZIO[R, IOException, Unit] =
-    source.toInputStream
-      .mapError(new IOException(_))
-      .use(
-        is =>
-          execute(_.storeFile(path, is))
-            .filterOrFail(identity)(InvalidPathError(s"Path is invalid. Cannot upload data to : $path"))
-            .unit
-      )
-
-  override def execute[T](f: JFTPClient => T): ZIO[Blocking, IOException, T] =
-    effectBlocking(f(unsafeClient)).refineToOrDie[IOException]
-}
-
-object Ftp {
-
-  def connect(settings: UnsecureFtpSettings): ZManaged[Blocking, ConnectionError, FtpClient[JFTPClient]] =
-    ZManaged.make(
-      effectBlocking {
-        val ftpClient = if (settings.secure) new JFTPSClient() else new JFTPClient()
-        settings.proxy.foreach(ftpClient.setProxy)
-        ftpClient.connect(settings.host, settings.port)
+import java.io.{File, FileOutputStream, IOException}
 
-        val success = ftpClient.login(settings.credentials.username, settings.credentials.password)
+import zio.blocking.Blocking
+import zio.nio.file.{Files, Path}
+import zio.stream.{ZSink, ZStream, ZStreamChunk}
+import zio.{ZIO, ZManaged}
 
-        if (settings.binary) {
-          ftpClient.setFileType(FTP.BINARY_FILE_TYPE)
-        }
+trait Ftp {
+  val ftp: Ftp.Service
+}
 
-        if (settings.passiveMode) {
-          ftpClient.enterLocalPassiveMode()
+object Ftp extends Serializable {
+
+  trait Service {
+    def connect[A](settings: FtpSettings[A]): ZManaged[Blocking, ConnectionError, FtpClient[A]]
```

Comment:
Move outside of `Service`, and put inside `Ftp` object directly:

```suggestion
    def connect[A](settings: FtpSettings[A]): ZLayer[Blocking, ConnectionError, FtpClient[A]]
```

Reply:
`Has[FtpClient[A]]`